vm/src/obj/objset.rs

```diff
@@ -56,536 +55,688 @@ impl PyValue for PyFrozenSet {
+        other =>  {return Err(vm.new_type_error(format!("{} is not a subtype of set or frozenset", other.class())));},
+        )
+    }
 
-    create_set(vm, elements, zelf.class())
-}
+    fn ge(self, other: PyObjectRef, vm: &VirtualMachine) -> PyResult {
+        match_class!(other,
+        set @ PySet => self.inner.borrow().ge(&set.inner.borrow(), vm),
+        frozen @ PyFrozenSet => self.inner.borrow().ge(&frozen.inner, vm),
+        other =>  {return Err(vm.new_type_error(format!("{} is not a subtype of set or frozenset", other.class())));},
+        )
+    }
 
-fn set_intersection(zelf: PyObjectRef, other: PyObjectRef, vm: &VirtualMachine) -> PyResult {
-    set_combine_inner(zelf, other, vm, SetCombineOperation::Intersection)
-}
+    fn gt(self, other: PyObjectRef, vm: &VirtualMachine) -> PyResult {
+        match_class!(other,
+        set @ PySet => self.inner.borrow().gt(&set.inner.borrow(), vm),
+        frozen @ PyFrozenSet => self.inner.borrow().gt(&frozen.inner, vm),
+        other =>  {return Err(vm.new_type_error(format!("{} is not a subtype of set or frozenset", other.class())));},
+        )
+    }
 
-fn set_difference(zelf: PyObjectRef, other: PyObjectRef, vm: &VirtualMachine) -> PyResult {
-    set_combine_inner(zelf, other, vm, SetCombineOperation::Difference)
-}
+    fn le(self, other: PyObjectRef, vm: &VirtualMachine) -> PyResult {
+        match_class!(other,
+        set @ PySet => self.inner.borrow().le(&set.inner.borrow(), vm),
+        frozen @ PyFrozenSet => self.inner.borrow().le(&frozen.inner, vm),
+        other =>  {return Err(vm.new_type_error(format!("{} is not a subtype of set or frozenset", other.class())));},
+        )
+    }
 
-fn set_symmetric_difference(
-    zelf: PyObjectRef,
-    other: PyObjectRef,
-    vm: &VirtualMachine,
-) -> PyResult {
-    validate_set_or_frozenset(vm, zelf.class())?;
-    validate_set_or_frozenset(vm, other.class())?;
-    let mut elements = HashMap::new();
-
-    for element in get_elements(&zelf).iter() {
-        let value = vm.call_method(&other, "__contains__", vec![element.1.clone()])?;
-        if !objbool::get_value(&value) {
-            elements.insert(element.0.clone(), element.1.clone());
-        }
+    fn lt(self, other: PyObjectRef, vm: &VirtualMachine) -> PyResult {
+        match_class!(other,
+        set @ PySet => self.inner.borrow().lt(&set.inner.borrow(), vm),
+        frozen @ PyFrozenSet => self.inner.borrow().lt(&frozen.inner, vm),
+        other =>  {return Err(vm.new_type_error(format!("{} is not a subtype of set or frozenset", other.class())));},
+        )
     }
 
-    for element in get_elements(&other).iter() {
-        let value = vm.call_method(&zelf, "__contains__", vec![element.1.clone()])?;
-        if !objbool::get_value(&value) {
-            elements.insert(element.0.clone(), element.1.clone());
-        }
+    fn union(self, other: PyObjectRef, vm: &VirtualMachine) -> PyResult {
```

Comment:
I think `other` should be a `PyIterable` since cpython accepts any iterable here:

```
>>> {1, 2, 3}.union([4])
{1, 2, 3, 4}
>>> {1, 2, 3}.union(4)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'int' object is not iterable
```

Reply:
Done